rust/src/nasl/builtin/network/socket.rs

```diff
@@ -534,81 +397,123 @@ impl NaslSockets {
-                        let fd =
-                            self.open_sock_tcp_tls(context, addr, port, bufsz, timeout, vhost)?;
-                        fds.push(self.add(fd))
+                    // IP
+                    Some(OpenvasEncaps::Ip) => {
+                        if let Ok(tcp) = TcpConnection::connect(
+                            addr,
+                            port,
+                            None,
+                            timeout,
+                            bufsz,
+                            get_retry(context),
+                        ) {
+                            // TODO: Set port transport
+                            tcp
+                        } else {
+                            continue;
+                        }
                     }
-                    _ => {
+                    // Unsupported transport layer
+                    None | Some(OpenvasEncaps::Max) => {
                         return Err(FunctionErrorKind::WrongArgument(format!(
-                            "unsupported transport layer: {transport}{tls_version}"
+                            "unsupported transport layer: {transport}(unknown)"
                         )))
                     }
-                },
-            }
+                    // TLS/SSL
+                    Some(tls_version) => match tls_version {
+                        OpenvasEncaps::Tls12 | OpenvasEncaps::Tls13 => {
+                            if let Ok(tls_conf) = Self::get_tls_conf(context) {
+                                if let Ok(tls) = create_tls_client(
+                                    vhost,
+                                    &tls_conf.cert_path,
+                                    &tls_conf.key_path,
+                                    &tls_conf.password,
+                                    &tls_conf.cafile_path,
+                                ) {
+                                    if let Ok(tcp) = TcpConnection::connect(
+                                        addr,
+                                        port,
+                                        Some(tls),
+                                        timeout,
+                                        bufsz,
+                                        get_retry(context),
+                                    ) {
+                                        tcp
+                                    } else {
+                                        continue;
+                                    }
+                                } else {
+                                    continue;
+                                }
+                            } else {
+                                continue;
+                            }
+                        }
+                        _ => {
+                            return Err(FunctionErrorKind::WrongArgument(format!(
+                                "unsupported transport layer: {transport}{tls_version}"
```

Comment:
q: although it is a wrong argument, that wrong argument could have been valid on a previously installed version (when e.g. a tls_version drops a transport layer, hence I think it should be a Diagnostic instead?